packages/core/helper-plugin/src/utils/hasPermissions.ts

```diff
@@ -16,6 +16,20 @@ const findMatchingPermissions = (
     );
 
     if (associatedPermission) {
+      // If there are locales defined on properties, then check user has permissions on that locale
+      if (
+        associatedPermission.properties?.locales &&
+        associatedPermission.properties?.locales.length > 0
+      ) {
+        const hasPermissionOnAtLeastOneLocale = associatedPermission.properties.locales.some(
+          (locale) => curr.properties?.locales?.includes(locale)
+        );
+
+        if (!hasPermissionOnAtLeastOneLocale) {
+          return acc;
+        }
+      }
```

Comment:
This isn't how this is done because you've now polluted the generic function with information from a plugin. In the i18n plugin we have redux middleware for the CM – I would suggest you do something similar